### PR TITLE
Add Solution and Tests for LeetCode Problem 844 - Backspace String Compare

### LINKED ISSUE #4: Problem 844 - Backspace String Compare
Problem 844 - Backspace String Compare

## Description
I would like to add a solution to the [844. Backspace String Compare](https://leetcode.com/problems/backspace-string-compare/) problem. The problem is about comparing two strings s and t after simulating a text editor's backspace functionality. The '#' character represents a backspace.

The problem has the following constraints:
- 1 <= s.length, t.length <= 200
- s and t only contain lowercase letters and '#' characters.

### Example 1:
Input: s = "ab#c", t = "ad#c"
Output: true
Explanation: Both s and t become "ac".

### Example 2:
Input: s = "ab##", t = "c#d#"
Output: true
Explanation: Both s and t become "".

### Example 3:
Input: s = "a#c", t = "b"
Output: false
Explanation: s becomes "c" while t becomes "b".

Follow up: Can you solve it in O(n) time and O(1) space?

## Task
- Implement a solution for the problem in Rust
- Add unit tests for the implemented solution
- Update the README.md file with a link to the problem and a brief description

### PR DESCRIPTION
**Pull request by DieHard073055**
## Problem Description
The problem, [Backspace String Compare](https://leetcode.com/problems/backspace-string-compare/), requires us to compare two strings after simulating a backspace operation. A '#' character represents a backspace operation. The goal is to determine if the two strings are equal after applying the backspace operations.

## Solution
In the provided solution, we use two `VecDeque` data structures, one for each input string. We iterate through the characters in each input string and perform the following steps:

1. If the current character is not '#', we push it to the front of the respective `VecDeque`.
2. If the current character is '#', we perform a `pop_front()` operation on the respective `VecDeque` to simulate the backspace operation.

Finally, we compare the two `VecDeque` structures to determine if they are equal.

## Tests
The solution includes a test module with six test cases, covering the examples from the problem statement and some additional edge cases. These tests help ensure that the `backspace_compare` function works correctly.

### CODE REVIEW

File: backspace-string-compare/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                s_stack.push_front(letter);
+            }
+        });
+        t.chars().for_each(|letter| {
+            if letter == '#' {
+                t_stack.pop_front();
+            } else {
+                t_stack.push_front(letter);
+            }
+        });
```
Comment (pamdayne) [inline backspace-string-compare/src/lib.rs]: L9-L15 and L16-L22 are repetitive. Let's implement DRY (don't repeat yourself) and use functions.

File: backspace-string-compare/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+            true
+        );
+        assert_eq!(
+            Solution::backspace_compare(String::from("ab##"), String::from("c#d#")),
+            true
+        );
+        assert_eq!(
+            Solution::backspace_compare(String::from("a#c"), String::from("b")),
+            false
+        );
```
Comment (pamdayne) [inline backspace-string-compare/src/lib.rs:41]: Should we cover more `false` scenarios?

Comment (DieHard073055) [reply backspace-string-compare/src/lib.rs:41]: It passed all the leetcode testcases. I think here we cover quite a decent amount of scenarios.

File: backspace-string-compare/src/lib.rs (hunk 1 above)
Comment (DieHard073055) [reply backspace-string-compare/src/lib.rs]: Thanks! done!

File: backspace-string-compare/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            }
+        });
+        stack
+    }
+    pub fn backspace_compare(s: String, t: String) -> bool {
+        let s_stack = Solution::eval_string(s);
+        let t_stack = Solution::eval_string(t);
+
+        s_stack == t_stack
+    }
```
Comment (pamdayne) [inline backspace-string-compare/src/lib.rs:21]: nice usage of function for both stacks!

**Review by pamdayne — APPROVED** [review]
